### PR TITLE
Enhance web response

### PR DESCRIPTION
**Pull request by Kolyunya**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13813

### CODE REVIEW

File: framework/web/Response.php [hunk 1]
```diff
@@ -292,13 +292,13 @@ public function setStatusCode($value, $text = null)
...
      * Sets the response status code based on the exception.
-     * @param \Exception $e
+     * @param \Throwable $e
      * @throws InvalidParamException if the status code is invalid.
      * @return $this the response object itself
      *
      * @since 2.0.12
      */
-    public function setStatusCodeByException(\Exception $e)
+    public function setStatusCodeByException(\Throwable $e)
```
Comment (samdark) [inline framework/web/Response.php]: Type hint needs to be removed since `\Throwable` is PHP 7 only.

**Review by samdark — CHANGES_REQUESTED** [review]

File: framework/web/Response.php (hunk 1 above)
Comment (Kolyunya) [inline framework/web/Response.php]: Fixed that. Do we revert docs fixes?

Comment (Kolyunya) [inline framework/web/Response.php]: Updated docs.

**Review by samdark — APPROVED** [review]

**Review by dynasource — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thanks!

**Comment by Kolyunya** [comment]
Don't you think that the changelog is a bit misleading? This seems like a new feature to me, not a compatibility fix according to the docs.

**Comment by samdark** [comment]
@Kolyunya you mean `UPGRADE`?

**Comment by Kolyunya** [comment]
I mean that according to the docs only `Exception` was supposed to be accepted by that method. Passing `Error` was not even documented. And now we added new functionality. This seems to be a new feature not a compatibility fix. I'm not sure if this is correct though.

**Comment by samdark** [comment]
Right. https://github.com/yiisoft/yii2/commit/da3bb5aa429f3f75af88694b25bb1e5b1ed97878

**Comment by Kolyunya** [comment]
I don't want to be annoying or anything but the signature of the existing method changed (without breaking the BC) and a new type of parameter is now accepted. It seems to me that the original changelog line is quite precise. Never mind anyway.

**Comment by rob006** [comment]
Tests are broken on PHP 5 - `Error` was introduced in PHP 7: https://travis-ci.org/yiisoft/yii2/jobs/214033905#L524

**Comment by Kolyunya** [comment]
Thanks! I made a patch https://github.com/yiisoft/yii2/pull/13834.

**Pull request merged**